core/src/main/java/com/netflix/conductor/core/events/EventProcessor.java

```diff
@@ -57,179 +56,152 @@
-				}catch(Exception e) {
-					payloadObj = payload;
-				}
-			}
-			
-			executionService.addMessage(queue.getName(), msg);
-			
-			String event = queue.getType() + ":" + queue.getName();
-			List<EventHandler> handlers = metadataService.getEventHandlersForEvent(event, true);
-			
-			for(EventHandler handler : handlers) {
-				
-				String condition = handler.getCondition();
-				logger.debug("condition: {}", condition);
-				if(!StringUtils.isEmpty(condition)) {
-					Boolean success = ScriptEvaluator.evalBool(condition, payloadObj);
-					if(!success) {
-						logger.info("handler {} condition {} did not match payload {}", handler.getName(), condition, payloadObj);
-						EventExecution ee = new EventExecution(msg.getId() + "_0", msg.getId());
-						ee.setCreated(System.currentTimeMillis());
-						ee.setEvent(handler.getEvent());
-						ee.setName(handler.getName());
-						ee.setStatus(Status.SKIPPED);
-						ee.getOutput().put("msg", payload);
-						ee.getOutput().put("condition", condition);
-						executionService.addEventExecution(ee);
-						continue;
-					}
-				}
-				
-				int i = 0;
-				List<Action> actions = handler.getActions();
-				for(Action action : actions) {
-					String id = msg.getId() + "_" + i++;
-					
-					EventExecution ee = new EventExecution(id, msg.getId());
-					ee.setCreated(System.currentTimeMillis());
-					ee.setEvent(handler.getEvent());
-					ee.setName(handler.getName());
-					ee.setAction(action.getAction());
-					ee.setStatus(Status.IN_PROGRESS);
-					if (executionService.addEventExecution(ee)) {
-						Future<Void> future = execute(ee, action, payload);
-						futures.add(future);
-					} else {
-						logger.warn("Duplicate delivery/execution? {}", id);
-					}
-				}
-			}
-			
-			for (Future<Void> future : futures) {
-				try {
-					future.get();
-				} catch (Exception e) {
-					logger.error(e.getMessage(), e);
-				}
-			}
-			
-			queue.ack(Arrays.asList(msg));
-			
-		} catch (Exception e) {
-			logger.error(e.getMessage(), e);
-		}
-	}
-
-	private Future<Void> execute(EventExecution ee, Action action, String payload) {
-		return executors.submit(()->{
-			try {
-				
-				logger.debug("Executing {} with payload {}", action.getAction(), payload);
-				Map<String, Object> output = actionProcessor.execute(action, payload, ee.getEvent(), ee.getMessageId());
-				if(output != null) {
-					ee.getOutput().putAll(output);
-				}
-				ee.setStatus(Status.COMPLETED);
-				executionService.updateEventExecution(ee);
-				
-				return null;
-			}catch(Exception e) {
-				logger.error(e.getMessage(), e);
-				ee.setStatus(Status.FAILED);
-				ee.getOutput().put("exception", e.getMessage());
-				return null;
-			}
-		});
-	}
-	
+    }
+
+    private void listen(ObservableQueue queue) {
+        queue.observe().subscribe((Message msg) -> handle(queue, msg));
+    }
+
+    private void handle(ObservableQueue queue, Message msg) {
+        try {
+            String payload = msg.getPayload();
```

Comment:
I see the payload's null check is removed. Can the payload be null? Or Would it break anything if it happens to be null?